src/scenes/level.js

```diff
@@ -255,6 +276,44 @@ export class LevelScene extends Phaser.Scene {
             this.autoClickers[i].release();
         }
         this.autoClickers = [];
-        this.clickObjects = [];
+        this.targets = [];
+    }
+
+    enemyKilled(name) {
+        // Update kill quest score
+        if (this.characterData[this.currentLevel].enemiesKilled[name] < this.killQuest) {
+            this.characterData[this.currentLevel].enemiesKilled[name]++;
+
+            let questCompleted = true;
+            this.targetMetaData.forEach((enemy, index) => {
+                // Check for quest completion
+                if (
+                    this.characterData[this.currentLevel].enemiesKilled[name] <
+                    this.killQuest
+                ) {
+                    questCompleted = false;
+                }
+                // Set as complete if all passed on last index
+                else if (questCompleted && index == this.targetMetaData.length - 1) {
+                    this.characterData[this.currentLevel].questCompleted = true;
+                    console.log("Quest complete!");
+                }
+            });
+        }
```

Comment:
What's the logic behind this function?

If you're just trying to check whether or not the quest has been completed, wouldn't something like this work: 

enemieskilled[name]++
if(enemiesKilled[name] == this.killQuest)
    characterData[currentLevel].questCompleted = true
    console.log("Quest complete")

What is the purpose of looping through each enemy in the targetMetaData object? Am just confused

Reply:
`killQuest` is an int for the number of enemies needed to be killed to finish the level. If there are multiple enemies you need to kill `killQuest` # of them for each. 

This loops through and checks if you have killed the needed amount for all enemies, if so you have passed the level.

Reply:
So you can only beat a level once you've killed a certain number of each mob? Or a total number of monsters killed?

Reply:
If the killQuest is 10 and the enemies are a cow and a goblin, you need to kill 10 cows and 10 goblins